src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -1621,8 +1862,10 @@ private static string ResolveEscapeFunction(string identifier, IEdmType bindingT
                 throw ExceptionUtil.CreateBadRequestError(ODataErrorStrings.RequestUriProcessor_EscapeFunctionMustHaveOneStringParameter(function.FullName()));
             }
 

```

Comment:
isn't this already validated in IsUrlEscapeFunction?  If so, maybe just assert here?

Reply:
Nope, IsUrlEscapeFunction just returns the functions based on the annotation. Although it is unlikely that this will ever get thrown if people annotate their escape functions carefully. 

---
In reply to: [369831446](https://github.com/OData/odata.net/pull/1621#discussion_r369831446) [](ancestors = 369831446)